databuilder/databuilder/extractor/snowflake_usage_extractor.py

```diff
@@ -0,0 +1,106 @@
+"""
+
+import logging
+from typing import (
+    Dict, Iterator, Union,
+)
+
+from pyhocon import ConfigFactory, ConfigTree
+from unidecode import unidecode
+
+from databuilder import Scoped
+from databuilder.extractor import sql_alchemy_extractor
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.sql_alchemy_extractor import SQLAlchemyExtractor
+from databuilder.models.table_column_usage import ColumnReader, TableColumnUsage
+
+LOGGER = logging.getLogger(__name__)
+
+class SnowflakeUsageExtractor(Extractor):
+    # SELECT statement from snowflake table that contains usage data
+    SQL_STATEMENT = """
+        SELECT
+            database,
+            schema,
+            name,
+            user_email,
+            read_count
+        FROM
+            {database}.{schema}.{table}
+        {where_clause_suffix};
+    """
+    # CONFIG KEYS
+    WHERE_CLAUSE_SUFFIX_KEY = 'where_clause_suffix'
+    POPULARITY_TABLE_DATABASE = 'popularity_table_database'
+    POPULARTIY_TABLE_SCHEMA = 'popularity_table_schema'
+    POPULARITY_TABLE_NAME = 'popularity_table_name'
+
+    DEFAULT_CONFIG = ConfigFactory.from_dict(
+        {WHERE_CLAUSE_SUFFIX_KEY: ' ',
+         POPULARITY_TABLE_DATABASE: 'PROD',
+         POPULARTIY_TABLE_SCHEMA: 'SCHEMA',
+         POPULARITY_TABLE_NAME: 'TABLE'}
+    )
+
+    def init(self, conf: ConfigTree) -> None:
+        conf = conf.with_fallback(SnowflakeUsageExtractor.DEFAULT_CONFIG)
+        self._where_clause_suffix=conf.get_string(SnowflakeUsageExtractor.WHERE_CLAUSE_SUFFIX_KEY)
+        self._popularity_table_database = conf.get_string(SnowflakeUsageExtractor.POPULARITY_TABLE_DATABASE)
+        self._popularity_table_schema = conf.get_string(SnowflakeUsageExtractor.POPULARTIY_TABLE_SCHEMA)
+        self._popularity_table_name = conf.get_string(SnowflakeUsageExtractor.POPULARITY_TABLE_NAME)
+
+        self.sql_stmt = SnowflakeUsageExtractor.SQL_STATEMENT.format(
+            where_clause_suffix=self._where_clause_suffix,
+            database=self._popularity_table_database,
+            schema=self._popularity_table_schema,
+            table=self._popularity_table_name
+        )
+
+        LOGGER.info("SQL for Snowflake popularity: {}".format(self.sql_stmt))
+
+        self._alchemy_extractor = sql_alchemy_extractor.from_surrounding_config(conf, self.sql_stmt)
+        sql_alch_conf = Scoped.get_scoped_conf(conf, self._alchemy_extractor.get_scope()) \
+            .with_fallback(ConfigFactory.from_dict({SQLAlchemyExtractor.EXTRACT_SQL: self.sql_stmt}))
+        self._alchemy_extractor.init(sql_alch_conf)
+        self._extract_iter: Union[None, Iterator] = None
+
+    def extract(self):
+        if not self._extract_iter:
+            self._extract_iter = self._get_extract_iter()
+        try:
+            return next(self._extract_iter)
+        except StopIteration:
+            return None
+
+    def get_scope(self) -> str:
+        return "extractor.snowflake"
```

Comment:
this scope should be unique to the extractor, I believe this one matches the snowflake metadata scope.